### PR TITLE
refactor(analytics): unify thumbnail placeholder, drop dead two-phase load

### PR DESCRIPTION
**Pull request by tphakala**
## What

Follow-up frontend cleanup after #3816/#3817 rerouted the analytics Species endpoints to the media proxy (defer-to-proxy). `/api/v2/analytics/species/summary` now returns a proxy `thumbnail_url` for every species.

1. **Remove the dead two-phase thumbnail loader** in `Species.svelte`. Because every species already carries a `thumbnail_url`, `loadThumbnailsAsync` filtered to nothing and never fetched `/species/thumbnails`. Deletes `loadThumbnailsAsync` and its `thumbnailFetchSeq` guard/call site (also removes a per-page-load burst of `ceil(N/20)` throttled fetches plus the re-render churn they caused). The `/api/v2/analytics/species/thumbnails` endpoint is **kept** as a standalone API; only its dead frontend caller is removed.

2. **Unify the missing-thumbnail placeholder** across the Species page family (`SpeciesCard`, `SpeciesCardMobile` (all 3 variants), `SpeciesDetailModal`, and the `Species.svelte` table) to the app-wide `handleBirdImageError` bird-silhouette, matching the dashboard and `AnalyticsOverview` (which already use it). Previously the Species surfaces used a divergent gray-box treatment that removed the `` on error via an `imageLoadFailed` state + a reset `$effect`. The silhouette keeps the `` (and its `alt`) and swaps `src` to `bird-placeholder.svg` on a proxy 404. Removing the `imageLoadFailed` state and reset effect simplifies the components; Svelte's reactive `src={species.thumbnail_url}` binding handles the reused-instance case (a changed species overrides the placeholder). The table `` gains the same `h-full w-full object-cover` sizing the sibling cards already use.

## Tests

- Updated the SpeciesCardMobile and SpeciesDetailModal error tests to assert the silhouette swap (img kept, `src` contains `bird-placeholder.svg`) instead of img removal.
- Removed the Species page test that exercised the deleted two-phase loader, plus the now-dead `/species/thumbnails` mocks.
- Added `SpeciesCard.test.ts` (the grid card previously had no test) covering the placeholder swap and the reactive rebind on species change.
- `npm run check:all` clean; full analytics suite green.

## Notes / follow-ups (pre-existing, filed separately)

- The visible change is the app-wide bird-silhouette placeholder on the Species page (previously a gray box), which matches the dashboard/overview. The placeholder SVG is not theme-aware (a light-gray patch in dark mode) app-wide; a dark-mode render check is worthwhile when that is addressed.
- Tracked separately: hardening `handleBirdImageError` against the placeholder asset itself failing, making the placeholder SVG theme-aware, and the analytics `{#each}` keys embedding the row index (defeats sort diffing). None are introduced by this PR.

## Summary by CodeRabbit

* **Bug Fixes**
  * Thumbnail images across analytics species UI now remain visible and swap to the bird placeholder when they fail to load.
  * Switching species after an error now updates the thumbnail correctly.
  * Added a guard to avoid repeated placeholder reload loops if the placeholder fails.
  * Improved the Species page request handling to prevent stale results; resetting filters no longer triggers an immediate refresh.

* **Refactor**
  * Centralized thumbnail error handling for consistent behavior across components.

* **Tests**
  * Updated and added tests to cover placeholder swap behavior, species switching, and the placeholder edge case.

### CODE REVIEW

File: frontend/src/lib/desktop/features/analytics/pages/Species.svelte [hunk 1]
```diff
@@ -202,13 +203,8 @@
     return (value * 100).toFixed(1) + '%';
   }
 
-  // Increments on every fetchData so an in-flight thumbnail loop from a previous
-  // fetch can detect it has been superseded and stop mutating speciesData.
-  let thumbnailFetchSeq = 0;
-
   async function fetchData() {
```
[the remark this answers is not in the document]
Comment (tphakala) [reply frontend/src/lib/desktop/features/analytics/pages/Species.svelte:211]: Good catch. I re-added a request sequence guard to fetchData in 45d5057. The old thumbnailFetchSeq went away with the dead two-phase loader, and the main summary fetch was never sequence-guarded, so a slow earlier response could clobber a newer one (and clear the loading flag out of order). Only the latest request now updates speciesData and isLoading.

File: frontend/src/lib/desktop/features/analytics/components/modals/SpeciesDetailModal.svelte [hunk 2]
```diff
@@ -98,14 +89,12 @@
...
-              alt={displayName}
-              class="w-full h-full object-cover"
-              onerror={handleImageError}
-            />
-          {/if}
+          <img
+            src={displaySpecies.thumbnail_url}
+            alt={displayName}
+            class="w-full h-full object-cover"
+            onerror={handleBirdImageError}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply frontend/src/lib/desktop/features/analytics/components/modals/SpeciesDetailModal.svelte:96]: Fixed in 45d5057, though with a re-entrancy guard rather than nulling target.onerror. Nulling onerror would permanently disable error handling on a reused  whose bound src later rebinds to a new (possibly also-failing) thumbnail. Instead I skip the swap when the current src is already the placeholder, which prevents the re-fire loop and keeps working for reused elements.

File: frontend/src/lib/desktop/features/analytics/components/ui/SpeciesCard.svelte [hunk 3]
```diff
@@ -26,30 +27,18 @@
...
-  });
 </script>
 
 <div class={cn('card bg-[var(--color-base-200)]', className)}>
   <figure class="px-4 pt-4">
     <div class="rounded-xl w-full aspect-[4/3] overflow-hidden bg-[var(--color-base-300)]">
-      {#if species.thumbnail_url && !imageLoadFailed}
+      {#if species.thumbnail_url}
         <img
           src={species.thumbnail_url}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply frontend/src/lib/desktop/features/analytics/components/ui/SpeciesCard.svelte:38]: Fixed in 45d5057. handleBirdImageError now skips the swap when src is already the placeholder, so a failing placeholder asset can't re-fire onerror in a loop. I used the src check instead of target.onerror = null so a reused  that rebinds to a new thumbnail still degrades correctly. Added a re-entrancy test.

**Comment by tphakala** [comment]
@coderabbitai review

File: frontend/src/lib/desktop/components/ui/image-utils.ts [hunk 4]
```diff
@@ -10,5 +10,11 @@ import { buildAppUrl } from '$lib/utils/urlHelpers';
  */
 export function handleBirdImageError(e: Event): void {
   const target = e.currentTarget as globalThis.HTMLImageElement;
-  target.src = buildAppUrl('/ui/assets/bird-placeholder.svg');
+  const placeholderSrc = buildAppUrl('/ui/assets/bird-placeholder.svg');
+  // Guard against an infinite onerror loop if the placeholder asset itself fails to
+  // load. Comparing against the current src (rather than nulling target.onerror)
+  // keeps error handling working for a reused <img> whose bound src later changes to
+  // a new (potentially also-failing) thumbnail.
+  if (target.src.endsWith(placeholderSrc)) return;
```
[the remark this answers is not in the document]
Comment (tphakala) [reply frontend/src/lib/desktop/components/ui/image-utils.ts]: Adopted includes() in b2c7ac4, matching the stable path so a query string or hash can't defeat the guard. Pulled the path into a BIRD_PLACEHOLDER_PATH constant so the check and the buildAppUrl call share one source of truth.

File: frontend/src/lib/desktop/features/analytics/components/modals/SpeciesDetailModal.svelte [hunk 5]
```diff
@@ -98,14 +89,12 @@
...
-              src={displaySpecies.thumbnail_url}
-              alt={displayName}
-              class="w-full h-full object-cover"
-              onerror={handleImageError}
-            />
-          {/if}
+          <img
+            src={displaySpecies.thumbnail_url}
+            alt={displayName}
+            class="w-full h-full object-cover"
```
[the remark this answers is not in the document]
Comment (tphakala) [reply frontend/src/lib/desktop/features/analytics/components/modals/SpeciesDetailModal.svelte:95]: This one doesn't apply here. thumbnail_url is a media-proxy URL keyed by scientific name (buildThumbnailURL -> ProxyImageURL(scientificName) on the server), so two distinct species never carry the same thumbnail_url string; the shared-URL precondition for this edge case can't occur with the analytics data.

The only case where Svelte skips the src update is reopening the SAME species whose image already 404'd, and there keeping the placeholder is the correct outcome. Re-adding the imageLoadFailed reset $effect would revert the simplification this PR makes and re-attempt a known-bad proxy URL on every reopen, which is exactly the churn defer-to-proxy is meant to avoid. Leaving it as-is.
